src/spac/transformations.py

```diff
@@ -100,6 +98,12 @@ def get_cluster_info(
     else:
         check_feature(adata, features=features)
 
+    # Convert data matrix or specified layer to DataFrame using to_df()
+    if layer:
+        data_df = adata.to_df(layer=layer)
+    else:
+        data_df = adata.to_df()
+
     # Check if the layer is specified and validate it
     if layer:

```

Comment:
@fangliu117, if you use adata.to_df(), why this part of the code is still there?